tools/cli/go-whisk/whisk/client.go

```diff
@@ -94,30 +94,22 @@ var DefaultObfuscateArr = []ObfuscateSet{
 
 func NewClient(httpClient *http.Client, config *Config) (*Client, error) {
 
-    // Disable certificate checking in the dev environment if in insecure mode
-    if config.Insecure {
-        Debug(DbgInfo, "Disabling certificate checking.\n")
-        var tlsConfig *tls.Config
-        if config.Cert != "" && config.Key != "" {
-            if cert, err := tls.LoadX509KeyPair(config.Cert, config.Key); err == nil {
-                tlsConfig = &tls.Config{
-                    Certificates: []tls.Certificate{cert},
-                    InsecureSkipVerify: true,
-                }
-            }
-        }else{
-            tlsConfig = &tls.Config{
-                InsecureSkipVerify: true,
-            }
-        }
+    if httpClient == nil {
+        httpClient = http.DefaultClient
+    }
 
-        http.DefaultClient.Transport = &http.Transport{
-            TLSClientConfig: tlsConfig,
+    tlsConfig := &tls.Config {
+        InsecureSkipVerify: config.Insecure,
+    }
+
+    if config.Cert != "" && config.Key != "" {
+        if cert, err := ReadX509KeyPair(config.Cert, config.Key); err == nil {
```

Comment:
should check for `err != nil` case here.. log error, return failure with message.

Reply:
We have a test case set the cert and key files via wsk property set --cert --key. If we raise an error if err is not nil, that test case will fail, because cert and key files are not available.

Reply:
i think we should log the failure at a minimum as this information might lead to the root cause of the eventual failure.

i'm still not yet convinced that the cli should continue if it was configured to use some specific certificate and that certificate cannot be read/used for some reason.

if the command is `wsk property`, then we can ignore the cert/key when creating the http client if both are not specified.

Reply:
This is already done by the function: func (c *Client) LoadX509KeyPair() error .